types/react-instantsearch-native/index.d.ts

```diff
@@ -60,6 +55,18 @@ export interface InstantSearchProps {
         props: any;
     };
 }
+
+export interface UsingSearchClient extends InstantSearchBaseProps {
+    searchClient: any;
+}
+
+export interface UsingManualInfo extends InstantSearchBaseProps  {
+    apiKey: string;
+    appId: string;
+    algoliaClient?: any;
```

Comment:
we are removing these in v6, so will be easy to update!

Reply:
I see. if that release is stable, I'll remove them asap.

Reply:
Oh, I misunderstood it, it is no need to input forever. OK I remove it,

Reply:
Please don't remove it yet!  Or if you do, then you should copy these types into a v5 folder as described here: https://github.com/DefinitelyTyped/DefinitelyTyped#if-a-library-is-updated-to-a-new-major-version-with-breaking-changes-how-should-i-update-its-type-declaration-package

This is important for all the people still using version 5.x of the react-instantsearch library who want to get upgrades or contribute back to the type definitions.

Reply:
@gburgett

https://github.com/algolia/react-instantsearch/commit/174cc28a6ed8ef09da1b5128c5d2157182303e3a

algoliaClient parameter deprecated from v5.1.0. If users need backward compatibility less than this version, I keep it, but I wonder if it's possible to avoid this by versioning as you described.

Reply:
And if it kept, I need some modification  because apiKey and appId exclusively independent on  algoliaClient. Please let me know if anyone has opinions about this.

Reply:
@tkow Thank you for pointing this out!  Yes it is deprecated, but deprecated is not the same thing as removed.  If we remove it from the types, anyone who consumes the types for v5 and uses the deprecated API will experience a *breaking change*.  Deprecation in general is used to avoid breaking changes, and warn the user that they will experience a breaking change *in the future*.

Since v6 of the react-instantsearch library is not out yet (current version on npmjs is 5.7.0), we should not make a separate v5 and v6 types folder.  The current types should continue to faithfully represent the public API of react-instantsearch 5.x.x, *including* deprecated properties.

If you would be willing to separate out the type definition then please do!  I am happy with wherever we land on this, so long as the end user can upgrade their type definitions and still use the deprecated API.

Reply:
@grurgett OK. I fixed it. This patch focus type mismatch so, backword compatibility type directory another PR.